client.go

```diff
@@ -1301,7 +1317,19 @@ func (c *HostClient) acquireConn() (*clientConn, error) {
 		return cc, nil
 	}
 	if !createConn {
-		return nil, ErrNoFreeConns
+		if !c.PreferWaitConn {
+			return nil, ErrNoFreeConns
+		}
+		timeout := time.Hour
+		if !deadline.IsZero() {
+			timeout = -time.Since(deadline)
```

Comment:
Lets not use an arbitrary timeout here of an hour. If `deadline` is zero it should just wait forever I guess. So in this if should be:
```go
<-c.connNotify:
return c.acquireConn(deadline)
```
and the other select in the `else`.

Reply:
thanks for you advise. i have ajust the implementation. when call clientDoDeadline , limit get conn with the deadline to avoid send request to server stil on getConn timeout. 
it's almost like my first version which no use time ticker.there is a little improve is that check timeout on every wakeup, to avoid make too much goroutines blocking.